src/clients/worker/worker.ts

```diff
@@ -399,6 +403,7 @@ export class Worker {
           this.logger.info(`Cancelled step run ${action.stepRunId}`);
         });
         future.cancel('Cancelled by worker');
+        await future.promise;
```

Comment:
not sure about this one. do we still want to wait on the cancelled step run promise?

Reply:
🤔 this is a bit of an issue i suppose, we cant handle cancelation because it will likely be invoked in a separate serverless context

Reply:
ok, not sure if I follow.. let's continue [here](https://github.com/hatchet-dev/hatchet-typescript/pull/106/files#r1635231318) as I needed this change in my other PR